hack/generated/pkg/util/armresourceresolver/resolver.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package armresourceresolver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/Azure/k8s-infra/hack/generated/pkg/genruntime"
+	"github.com/Azure/k8s-infra/hack/generated/pkg/util/kubeclient"
+)
+
+type Resolver struct {
+	client *kubeclient.Client
+}
+
+func NewResolver(client *kubeclient.Client) *Resolver {
+	return &Resolver{
+		client: client,
+	}
+}
+
+// GetResourceGroupAndFullAzureName gets the full name for use in creating a resource. This name includes
+// the full "path" to the resource being deployed. For example, a Virtual Network Subnet's name might be:
+// "myvnet/mysubnet"
+func (r *Resolver) GetResourceGroupAndFullAzureName(ctx context.Context, obj genruntime.MetaObject) (string, string, error) {
+	owner := obj.Owner()
+
+	if obj.GetObjectKind().GroupVersionKind().Kind == "ResourceGroup" {
+		return getAzureName(obj), "", nil
+	}
+
+	if owner == nil {
+		return "", "", errors.New(fmt.Sprintf(
+			"Can't GetOwnerAndResourceGroupDetails from %s (kind: %s), which has no owner but is not a ResourceGroup",
+			obj.GetName(),
+			obj.GetObjectKind().GroupVersionKind()))
+	}
+
+	// TODO: This is a hack for now since we don't have an RG type yet
+	if owner.Kind == "ResourceGroup" {
+		return owner.Name, getAzureName(obj), nil
+	}
+
+	var ownerGvk schema.GroupVersionKind
+	found := false
+	// TODO: We need to find the specific storage version GVK...
+	for gvk := range r.client.Scheme.AllKnownTypes() {
+		if gvk.Group == owner.Group && gvk.Kind == owner.Kind {
+			if !found {
+				ownerGvk = gvk
+				found = true
+			} else {
+				return "", "", errors.Errorf("owner group: %s, kind: %s has multiple possible schemes registered", owner.Group, owner.Kind)
+			}
+		}
+	}
+
+	// TODO: We could do this on launch probably since we can check based on the AllKnownTypes() collection
```

Comment:
```suggestion
	// TODO: We should do this on launch since we can check based on the `AllKnownTypes()` collection
```

😝

Reply:
I think I agree with could -> should but disagree with the backticks - this isn't markdown and other places in the code we just refer to functions/classes by name without backticks.

Reply:
Whoops. The backticks were a reflex while typing! Change looks good.